[processor/k8sattribute] e2e test fails to setup pod in required time limit

### Component(s)

processor/k8sattributes

### What happened?

## Description

Logs from test failure:

```
=== RUN   TestE2E
    k8s_telemetrygen.go:64: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/k8stest/k8s_telemetrygen.go:64
        	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/processor/k8sattributesprocessor/e2e_test.go:75
        	Error:      	Condition never satisfied
        	Test:       	TestE2E
        	Messages:   	telemetrygen pod of Workload [Job] in datatype [logs] haven't started within 3 minutes, latest pod phase is 
W0710 [14](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5509366365/jobs/10042200389?pr=24065#step:11:15):50:11.23[17](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5509366365/jobs/10042200389?pr=24065#step:11:18)66   17282 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0710 14:50:11.427792   17282 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
W0710 14:50:11.628093   17282 warnings.go:70] child pods are preserved by default when jobs are deleted; set propagationPolicy=Background to remove them or set propagationPolicy=Orphan to suppress this warning
--- FAIL: TestE2E ([19](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5509366365/jobs/10042200389?pr=24065#step:11:20)7.01s)
```
[Test failure link](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5509366365/jobs/10042200389?pr=24065)

This occurred on an unrelated [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24065), so I assume it's a flaky test and just wasn't able to set up the environment within the time limit. I have not investigated further to confirm though.

### Collector version

v0.81.0

### Environment information

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")


### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

--- a/.github/workflows/e2e-tests.yml
+++ b/.github/workflows/e2e-tests.yml
@@ -32,15 +32,23 @@ jobs:
         run: make -j2 gomoddownload
       - name: Build Docker Image
         run: |
-          make docker-otelcontribcol
-      - name: export image to tar
+          make docker-otelcontribcol docker-telemetrygen
+      - name: export otelcontribcol image to tar
         run: |
           docker save otelcontribcol:latest > /tmp/otelcontribcol.tar
-      - name: Upload artifact
+      - name: export telemetrygen image to tar
+        run: |
+          docker save telemetrygen:latest > /tmp/telemetrygen.tar
+      - name: Upload otelcontribcol artifact
         uses: actions/upload-artifact@v3
         with:
           name: otelcontribcol
           path: /tmp/otelcontribcol.tar
+      - name: Upload telemetrygen artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: telemetrygen
+          path: /tmp/telemetrygen.tar
   kubernetes-test:
     env:
       KUBECONFIG: /tmp/kube-config-otelcol-e2e-testing
@@ -78,12 +86,23 @@ jobs:
         with:
           name: otelcontribcol
           path: /tmp
-      - name: Load Docker image
+      - name: Load otelcontribcol Docker image
         run: |
           docker load --input /tmp/otelcontribcol.tar
       - name: Kind load image
         run: |
           kind load docker-image otelcontribcol:latest --name kind
+      - name: Download artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: telemetrygen
+          path: /tmp
+      - name: Load telemetrygen Docker image
+        run: |
+          docker load --input /tmp/telemetrygen.tar
+      - name: Kind load image
+        run: |
+          kind load docker-image telemetrygen:latest --name kind
       - name: run k8sclusterreceiver e2e tests
         run: |
           cd receiver/k8sclusterreceiver

--- a/processor/k8sattributesprocessor/e2e_test.go
+++ b/processor/k8sattributesprocessor/e2e_test.go
@@ -54,8 +54,9 @@ func newExpectedValue(mode int, value string) *expectedValue {
 // `/tmp/kube-config-otelcol-e2e-testing`. Run the following command prior to running the test locally:
 //
 //	kind create cluster --kubeconfig=/tmp/kube-config-otelcol-e2e-testing
-//	make docker-otelcontribcol
+//	make docker-otelcontribcol docker-telemetrygen
 //	KUBECONFIG=/tmp/kube-config-otelcol-e2e-testing kind load docker-image otelcontribcol:latest
+//	KUBECONFIG=/tmp/kube-config-otelcol-e2e-testing kind load docker-image telemetrygen:latest
 func TestE2E(t *testing.T) {
 	kubeConfig, err := clientcmd.BuildConfigFromFlags("", testKubeConfig)
 	require.NoError(t, err)
@@ -104,7 +105,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-traces-job"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},
@@ -126,7 +127,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-traces-statefulset"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},
@@ -150,7 +151,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-traces-deployment"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},
@@ -172,7 +173,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-traces-daemonset"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},
@@ -194,7 +195,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-metrics-job"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},
@@ -216,7 +217,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-metrics-statefulset"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},
@@ -240,7 +241,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-metrics-deployment"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},
@@ -262,7 +263,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-metrics-daemonset"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},
@@ -284,7 +285,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-logs-job"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},
@@ -306,7 +307,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-logs-statefulset"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},
@@ -330,7 +331,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-logs-deployment"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},
@@ -352,7 +353,7 @@ func TestE2E(t *testing.T) {
 				"k8s.labels.app":           newExpectedValue(equal, "telemetrygen-"+testID+"-logs-daemonset"),
 				"k8s.container.name":       newExpectedValue(equal, "telemetrygen"),
 				"k8s.cluster.uid":          newExpectedValue(exist, ""),
-				"container.image.name":     newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.name":     newExpectedValue(equal, "telemetrygen"),
 				"container.image.tag":      newExpectedValue(equal, "latest"),
 				"container.id":             newExpectedValue(exist, ""),
 			},

--- a/processor/k8sattributesprocessor/testdata/e2e/telemetrygen/daemonset.yaml
+++ b/processor/k8sattributesprocessor/testdata/e2e/telemetrygen/daemonset.yaml
@@ -24,6 +24,7 @@ spec:
         - --rate=1
         - --otlp-attributes=service.name="test-{{ .DataType }}-daemonset"
         - --otlp-attributes=k8s.container.name="telemetrygen"
-        image: ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:latest
+        image: telemetrygen:latest
+        imagePullPolicy: Never
         name: telemetrygen
       restartPolicy: Always

--- a/processor/k8sattributesprocessor/testdata/e2e/telemetrygen/deployment.yaml
+++ b/processor/k8sattributesprocessor/testdata/e2e/telemetrygen/deployment.yaml
@@ -25,7 +25,7 @@ spec:
         - --rate=1
         - --otlp-attributes=service.name="test-{{ .DataType }}-deployment"
         - --otlp-attributes=k8s.container.name="telemetrygen"
-        image: ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:latest
-        imagePullPolicy: IfNotPresent
+        image: telemetrygen:latest
+        imagePullPolicy: Never
         name: telemetrygen
       restartPolicy: Always

--- a/processor/k8sattributesprocessor/testdata/e2e/telemetrygen/job.yaml
+++ b/processor/k8sattributesprocessor/testdata/e2e/telemetrygen/job.yaml
@@ -21,6 +21,7 @@ spec:
         - --duration=36000s
         - --otlp-attributes=service.name="test-{{ .DataType }}-job"
         - --otlp-attributes=k8s.container.name="telemetrygen"
-        image: ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:latest
+        image: telemetrygen:latest
+        imagePullPolicy: Never
         name: telemetrygen
-      restartPolicy: Never
+      restartPolicy: OnFailure

--- a/processor/k8sattributesprocessor/testdata/e2e/telemetrygen/statefulset.yaml
+++ b/processor/k8sattributesprocessor/testdata/e2e/telemetrygen/statefulset.yaml
@@ -26,6 +26,7 @@ spec:
         - --rate=1
         - --otlp-attributes=service.name="test-{{ .DataType }}-statefulset"
         - --otlp-attributes=k8s.container.name="telemetrygen"
-        image: ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:latest
+        image: "telemetrygen:latest"
+        imagePullPolicy: Never
         name: telemetrygen
       restartPolicy: Always

--- a/receiver/k8sclusterreceiver/e2e_test.go
+++ b/receiver/k8sclusterreceiver/e2e_test.go
@@ -36,8 +36,9 @@ const testKubeConfig = "/tmp/kube-config-otelcol-e2e-testing"
 // `/tmp/kube-config-otelcol-e2e-testing`. Run the following command prior to running the test locally:
 //
 //	kind create cluster --kubeconfig=/tmp/kube-config-otelcol-e2e-testing
-//	make docker-otelcontribcol
+//	make docker-otelcontribcol docker-telemetrygen
 //	KUBECONFIG=/tmp/kube-config-otelcol-e2e-testing kind load docker-image otelcontribcol:latest
+//	KUBECONFIG=/tmp/kube-config-otelcol-e2e-testing kind load docker-image telemetrygen:latest
 func TestE2E(t *testing.T) {
 	var expected pmetric.Metrics
 	expectedFile := filepath.Join("testdata", "e2e", "expected.yaml")

